### PR TITLE
#160452446 In-app notification

### PR DESCRIPTION
**Pull request by hnobi**
#### What does this PR do?
 Add real-time in-app notification functionality
#### Description of Task to be completed?
-> Adds reducer and actions for fetching in-app notifications
-> Adds reducer and actions for updating  notifications
-> integrates socket.io for real-time notification
-> Adds a drop-down for displaying notifications
-> Adds test for notification reducers
->  Adds test for notification actions

#### How should this be manually tested?

```
git checkout staging 
```
```
 git pull
```
```
git checkout ft-160452446-view-notification
```
```
npm install
```
Register Author's Haven and login and you will see notification icon on the navbar
#### Any background context you want to provide?
Nil
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160452446

#### Screenshots (if appropriate)
a

#### Questions:
Nil

### CODE REVIEW

File: src/components/header/NavBarDefault.jsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
+import DropDown from '../inAppNotification/notification/dropDown';
+// import AppNotificatonContainer from '../inAppNotification/AppNotificatonContainer';
```
Comment (blackshady) [inline src/components/header/NavBarDefault.jsx:5]: Nice job on this PR. Please take out the commented code if you no longer need them

File: src/components/inAppNotification/notification/style.scss [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    padding: 0;
+    top: 46px;
+    transform: translate(-50%, 0);
+    background: white;
+    min-height: 10px;
+    min-width: 10px;
+    width: 300px;
+    z-index: 150;
+    //-webkit-box-shadow: 0 1px 2px rgba(0,0,0,.25), 0 0 1px rgba(0,0,0,.35);
+    //box-shadow: 0 1px 2px rgba(0,0,0,.25), 0 0 1px rgba(0,0,0,.35);
```
Comment (blackshady) [inline src/components/inAppNotification/notification/style.scss]: Please also take this commented line of code.

File: src/components/landingPage/LandingPageContainer.jsx [hunk 3]
```diff
@@ -43,6 +44,10 @@ export class LandingPageContainer extends React.Component {
     const urlParams = new URLSearchParams(this.props.location.search);
     const login = urlParams.get('login');
     const register = urlParams.get('register');
+    // setUserId(this.props.userId);
```
Comment (blackshady) [inline src/components/landingPage/LandingPageContainer.jsx]: Please also take out this commented code if you no longer need them.

File: src/redux/reducer/articles.js [hunk 4]
```diff
@@ -82,8 +97,17 @@ const articleReducer = (state = defaultState, action) => {
...
+        loading: false,
+        open: false,
+        errors: response,
+        statusCode
       };
+    // case GET_COMMENTS:
+    //   return {
+    //     ...state,
+    //     comment: commentsReducer(state.comment, action)
+    //   };
```
Comment (blackshady) [inline src/redux/reducer/articles.js]: Please also take out the commented code if you no longer need them.

File: src/components/header/NavBarDefault.jsx (hunk 1 above)
Comment (hnobi) [reply src/components/header/NavBarDefault.jsx:5]: Thanks for your feedback. I will remove all commented codes now

File: src/services/socket.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+// import { getUserId } from '../redux/actions/inAppNotification';
+
+// const socket = openSocket('odin-ah-backend-staging.herokuapp.com/api/v1');
+// const userId = getUserId();
+// console.log(userId, 'socket>>>>>>>>>>>>>>.');
+// const connect = () => {
+//   socket.on(`notification-${userId}`, data => console.log('data', data));
+// };
+
+// export { connect };
```
Comment (blackshady) [inline src/services/socket.js]: Please remove this file if you no longer need them.

File: src/tests/components/notifications/displayIconView.test.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import DisplayIconView from '../../../components/inAppNotification/notification/displayIconView';
+
+describe('displayIconView', () => {
+  // const props = {
+  //   totalNotification: 3,
+  // };
```
Comment (blackshady) [inline src/tests/components/notifications/displayIconView.test.js]: Please also take out the commented code if you no longer need them.

**Review by uchemukolo — COMMENTED** [review]
Great job @hnobi for implementing this feature. I noticed your test is below the 80% threshold set. Try and improve the coverage. Also, update the screenshot for the notification dropdown to reflect the updated changes after implementing feedback from yesterday.

File: src/components/landingPage/LandingPageContainer.jsx (hunk 3 above)
Comment (hnobi) [reply src/components/landingPage/LandingPageContainer.jsx]: Thanks for this feedback, I have removed it.

**Review by waleCloud — APPROVED** [review]

**Pull request merged**